ngraph/frontend/paddlepaddle/src/op/interp.cpp

```diff
@@ -22,6 +22,58 @@ namespace frontend {
 namespace pdpd {
 namespace op {
 
+std::shared_ptr<ngraph::Node>
+calculate_output_shape_based_on_scales(const Output<ngraph::Node>& data,
+                                       const std::vector<float>& scale,
+                                       Output<ngraph::Node>& scales)
+{
+    if (scale.size() == 1)
+        scales = opset6::Constant::create<float>(element::f32, Shape{4}, {1, 1, scale[0], scale[0]});
+    else if (scale.size() == 2)
+        scales = opset6::Constant::create<float>(element::f32, Shape{4}, {1, 1, scale[0], scale[1]});
+    else if (scale.size() == 3)
+        scales = opset6::Constant::create<float>(element::f32, Shape{4}, {1, scale[0], scale[1], scale[2]});
+    else if (scale.size() == 4)
```

Comment:
If scale.size() > 5 or size==0 - should we raise some error? If not, maybe just add 'assert' at the beginning Also this method is completely not covered by unit tests - is it needed?

Reply:
will handle the corner issue by adding ASSERT

Reply:
fix in 0f9058cb8edab2b1cd0b740e8717ac6582eca6a3